test/components/opt-in-flyout/opt-in-flyout.test.js

```diff
@@ -0,0 +1,183 @@
+			expect(title.textContent).to.equal('Flyout Demo');
+		});
+
+		it('should reflect title attribute to property', async() => {
+			const newTitle = 'new title';
+			flyout.setAttribute('title', newTitle);
+			await flyout.updateComplete;
+			await innerFlyout.updateComplete;
+			expect(flyout.title).to.equal(newTitle);
+			const title = innerFlyout.shadowRoot.querySelector('#title');
+			expect(title.textContent).to.equal(newTitle);
+		});
+
+		it('should contain long description', () => {
+			const longDescription = innerFlyout.shadowRoot.querySelector('#long-description');
+			expect(longDescription).to.exist;
+		});
+
+		it('should contain tutorial link', () => {
+			const tutorial = innerFlyout.shadowRoot.querySelector('.flyout-tutorial');
+			const links = TestUtil.selectVisible(tutorial, 'a');
+
+			expect(links.length).to.equal(2);
+
+			expect(links[0].href).to.contain('https://www.testlink1.com');
+			expect(links[0].textContent).to.contain('tutorials');
+		});
+
+		it('should contain help documentation link', () => {
+			const tutorial = innerFlyout.shadowRoot.querySelector('.flyout-tutorial');
+			const links = TestUtil.selectVisible(tutorial, 'a');
+
+			expect(links.length).to.equal(2);
+
+			expect(links[1].href).to.contain('https://www.testlink2.com');
+			expect(links[1].textContent).to.contain('help documentation');
+
+		});
+
+		it('should contain only tutorial specific text when only tutorial link specified', async() => {
+			innerFlyout.helpDocsLink = null;
+			await innerFlyout.updateComplete;
+			const tutorial = innerFlyout.shadowRoot.querySelector('.flyout-tutorial');
+			const links = TestUtil.selectVisible(tutorial, 'a');
+
+			expect(links.length).to.equal(1);
+
+			expect(links[0].href).to.contain('https://www.testlink1.com');
+			expect(links[0].textContent).to.contain('tutorials');
+
+		});
+
+		it('should contain only help documentation specific text when only help link specified', async() => {
+			innerFlyout.tutorialLink = null;
+			await innerFlyout.updateComplete;
+			const tutorial = innerFlyout.shadowRoot.querySelector('.flyout-tutorial');
+			const links = TestUtil.selectVisible(tutorial, 'a');
+
+			expect(links.length).to.equal(1);
+
+			expect(links[0].href).to.contain('https://www.testlink2.com');
+			expect(links[0].textContent).to.contain('help documentation');
+		});
+
+		it('should contain enable button', () => {
+			const button = innerFlyout.shadowRoot.querySelector('d2l-button[primary]');
+			expect(button.textContent).to.equal('Turn it on');
+			expect(button).to.exist;
+		});
+
+		it('should contain leave it off button', () => {
+			const button = innerFlyout.shadowRoot.querySelector('d2l-button:not([primary])');
+			expect(button.textContent).to.equal('Leave it off');
+			expect(button).to.exist;
+		});
+
+		it('should fire opt-in event when enable button clicked', async() => {
+			const button = innerFlyout.shadowRoot.querySelector('d2l-button[primary]');
+			clickElem(button);
```

Comment:
I'm guessing that `clickElem` has a `setTimeout` or rAF baked into it somewhere to ensure it runs after the event wire-up, ya?

Reply:
Yeah exactly -- we've done a bunch of testing with these and confirmed that all the interaction events have sufficient timeouts. Behind the scenes, this is communicating via web-socket back to the server, which then instructions the browser to perform the task, which then communicates back.

Reply:
On one hand, convenient. On the other, feels weird... I'm always going to be thinking to myself that all the clicking code will run before the event wire-up. I do worry a little about confusion if people aren't sure when they need to explicitly delay (using `setTimeout` or getting the promise and returning it) vs just trusting the helper to be delayed. I suppose if they are always using this helper or a helper that works in the same way, then there shouldn't be any need to write it differently, and this no source of confusion. 🤷‍♂️